www/base/src/app/common/directives/buildsticker/buildsticker.directive.spec.coffee

```diff
@@ -0,0 +1,54 @@
+beforeEach module 'app'
+
+describe 'buildsticker controller', ->
+    buildbotService = mqService = $httpBackend = $rootScope = $compile = null
+
+    injected = ($injector) ->
+        $compile = $injector.get('$compile')
+        $httpBackend = $injector.get('$httpBackend')
+        $location = $injector.get('$location')
+        decorateHttpBackend($httpBackend)
+        $rootScope = $injector.get('$rootScope')
+        mqService = $injector.get('mqService')
+        $controller = $injector.get('$controller')
+        $q = $injector.get('$q')
+
+        # stub out the actual backend of mqservice
+        spyOn(mqService,"setBaseUrl").and.returnValue(null)
+        spyOn(mqService,"startConsuming").and.returnValue($q.when( -> ))
+        spyOn(mqService,"stopConsuming").and.returnValue(null)
+        buildbotService = $injector.get('buildbotService')
+
+    beforeEach(inject(injected))
+
+    afterEach ->
+        $httpBackend.verifyNoOutstandingExpectation()
+        $httpBackend.verifyNoOutstandingRequest()
+
+    it 'directive should generate correct html', ->
+        $httpBackend.expectDataGET('builds/1')
+        buildbotService.one('builds', 1).bind($rootScope)
+        $httpBackend.flush()
+        $httpBackend.expectDataGET('builders/1')
+        element = $compile("<buildsticker build='build'></buildsticker>")($rootScope)
+        $httpBackend.flush()
+        $rootScope.build.started_at = moment.unix() # avoid test failed due to time changes
+        $rootScope.$digest()
+        html = element.html()
+
+        expectHtml = '''
+        <div class="panel-heading no-select">
```

Comment:
I would prefer to avoid testing raw html like this.

Rather make sure that some imporant characteristics are found (href is created by ui-sref, that kind of thing)

This will avoid to update the test for trivial template changes